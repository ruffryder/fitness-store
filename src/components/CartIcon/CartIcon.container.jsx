import React from "react";
import { connect } from "react-redux";
import { toggleCartHidden } from "../../redux/cart/cart.actions";
import { selectCartItemsCount } from "../../redux/cart/cart.selectors";
import { createStructuredSelector } from "reselect";
import CartIcon from "./CartIcon";

function CartIconContainer({ toggleCart, itemCount }) {
  return <CartIcon toggleCart={toggleCart} itemCount={itemCount} />;
}

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount
});

const mapDispatchToProps = dispatch => ({
  toggleCart: () => dispatch(toggleCartHidden())
});

export default connect(mapStateToProps, mapDispatchToProps)(CartIconContainer);
