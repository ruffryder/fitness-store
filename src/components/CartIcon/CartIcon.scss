@import "../../variables.scss";

.cart-icon {
  color: $primary-color;
  width: 55px;
  height: 55px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  transition: all 0.2s ease-in;
  .shopping-icon {
    width: 30px;
    height: 30px;
  }

  .item-count {
    position: absolute;
    font-size: 12px;
    font-weight: bold;
    bottom: 38px;
    right: 8px;
  }
  &:hover {
    transform: scale(1.15);
  }
}
