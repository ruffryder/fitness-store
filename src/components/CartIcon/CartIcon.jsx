import React from "react";
import { FaShoppingCart } from "react-icons/fa";
import "./CartIcon.scss";

function CartIcon({ toggleCart, itemCount }) {
  return (
    <div className="cart-icon" onClick={toggleCart}>
      <FaShoppingCart className="shopping-icon" />
      <span className="item-count">{itemCount}</span>
    </div>
  );
}

export default CartIcon;
