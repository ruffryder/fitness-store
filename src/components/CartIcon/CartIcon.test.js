import { shallow } from "enzyme";
import React from "react";
import CartIcon from "./CartIcon";

let wrapper;

beforeEach(() => {
  const mockProps = {
    toggleCart: jest.fn(),
    itemCount: 0
  };
  wrapper = shallow(<CartIcon {...mockProps} />);
});

describe("Cart Icon functionality", () => {
  it("renders CartIcon component without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
