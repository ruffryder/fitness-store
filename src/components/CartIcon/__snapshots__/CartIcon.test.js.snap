// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Icon functionality renders CartIcon component without crashing 1`] = `ShallowWrapper {}`;
